api/src/main/java/org/openmrs/Diagnosis.java

```diff
@@ -0,0 +1,159 @@
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+/**
+ * Diagnosis class defines the identification of the nature of an illness or other problem 
+ * by examination of the symptoms during an encounter (visit or interaction of a patient with a healthcare worker).
+ */
+
+public class Diagnosis extends BaseChangeableOpenmrsData{
+	
+	private Integer diagnosisId;
+	
+	private Encounter encounter;
+	
+	private CodedOrFreeText diagnosis;
+	
+	private Condition condition;
+
+	private ConditionVerificationStatus certainty;
+	
+	private Integer rank;
+	
+	public Diagnosis() {
+	}
+
+	/**
+	 * Diagnosis constructor with encounter, CodedOrFreeText, ConditionVerificationStatus, and rank
+	 *
+	 * @param encounter the encounter during which the diagnosis was made
+	 * @param diagnosis the CodedOrFreeText of the diagnosis
+	 * @param certainty the ConditionVerificationStatus of the diagnosis
+	 * @param rank the rank of the diagnosis
+	 */
+	
+	public Diagnosis(Encounter encounter, CodedOrFreeText diagnosis, ConditionVerificationStatus certainty, Integer rank) {
+		this.encounter = encounter;
+		this.diagnosis = diagnosis;
+		this.certainty = certainty;
+		this.rank = rank;
+	}
+	
+	/**
+	 * Gets the diagnosisId.
+	 * 
+	 * @return diagnosisId
+	 */
+	@Override
+	public Integer getId() {
+		return diagnosisId;
+	}
+
+	/**
+	 * Sets diagnosis diagnosisId
+	 * 
+	 * @param diagnosisId
+	 */
+	@Override
+	public void setId(Integer diagnosisId) {
+		this.diagnosisId = diagnosisId;
+	}
+
+	/**
+	 * Gets the encounter associated with this diagnosis.
+	 * 
+	 * @return encounter associated with this diagnosis
+	 */
+	public Encounter getEncounter() {
+		return encounter;
+	}
+
+	/**
+	 * Sets the encounter associated with this diagnosis.
+	 * 
+	 * @param encounter the encounter object when this diagnosis was made
+	 */
+	public void setEncounter(Encounter encounter) {
+		this.encounter = encounter;
+	}
+
+	/**
+	 * Gets the diagnosis.
+	 * 
+	 * @return diagnosis the CodedOrFreeText associated with this diagnosis
+	 */
+	public CodedOrFreeText getDiagnosis() {
+		return diagnosis;
+	}
+
+	/**
+	 * Sets the CodedOrFreeText diagnosis.
+	 * 
+	 * @param diagnosis the CodedOrFreeText associated with this diagnosis
+	 */
+	public void setDiagnosis(CodedOrFreeText diagnosis)  {
+		this.diagnosis = diagnosis;
+	}
+
+	/**
+	 * Gets the diagnosis certainty.
+	 * 
+	 * @return certainty the ConditionVerificationStatus of this Diagnosis
+	 */
+	public ConditionVerificationStatus getCertainty() {
+		return certainty;
+	}
+
+	/**
+	 * Sets the diagnosis certainty
+	 * 
+	 * @param certainty 
+	 */
+	public void setCertainty(ConditionVerificationStatus certainty)  {
+		this.certainty = certainty;
+	}
+
+	/**
+	 * Gets the diagnosis rank.
+	 * 
+	 * @return rank 
+	 */
+	public Integer getRank() {
+		return rank;
+	}
+
+	/**
+	 * Sets diagnosis rank
+	 * 
+	 * @param rank
```

Comment:
You also need to document the params and return values. For examples, take a look at https://github.com/openmrs/openmrs-core/commit/f5132a1dda4a6a8fcbfe06a32dcbbe4b7a0a270d